t_frang/test_length.py

```diff
@@ -0,0 +1,374 @@
+        },
+        {
+            'id': 'curl-2',
+            'type': 'external',
+            'binary': 'curl',
+            'cmd_args': '-Ikf -v http://127.0.0.4:8765/ -H "Host: tempesta-tech.com:8765"  -H "X-Long: {0}"'.format( 
+                '1' * 293,
+            ),
+        },
+        {
+            'id': 'curl-22',
+            'type': 'external',
+            'binary': 'curl',
+            'cmd_args': '-Ikf -v http://127.0.0.4:8765/ -H "Host: tempesta-tech.com:8765"',
+        },
+        {
+            'id': 'curl-3',
+            'type': 'external',
+            'binary': 'curl',
+            'cmd_args': '-kf -v http://127.0.0.4:8765/ -H "Host: tempesta-tech.com:8765" -d {0}'.format(  
+                {'some_key_long_one': 'some_value'},
+            ),
+        },
+        {
+            'id': 'curl-31',
+            'type': 'external',
+            'binary': 'curl',
+            'cmd_args': '-kf -v http://127.0.0.4:8765/ -H "Host: tempesta-tech.com:8765"'
+        },
+        {
+            'id': 'curl-32',
+            'type': 'external',
+            'binary': 'curl',
+            'cmd_args': '-kf -v http://127.0.0.4:8765/ -H "Host: tempesta-tech.com:8765" -d {0}'.format(  
+                {'12345678': '1'},
+            ),
+        },
+    ]
+
+    tempesta = {
+        'config': """
+            frang_limits {
+                http_uri_len 5;
+                http_field_len 300;
+                http_body_len 10;
+            }
+
+            listen 127.0.0.4:8765;
+
+            srv_group default {
+                server ${server_ip}:8000;
+            }
+
+            vhost tempesta-cat {
+                proxy_pass default;
+            }
+
+            tls_match_any_server_name;
+            tls_certificate RSA/tfw-root.crt;
+            tls_certificate_key RSA/tfw-root.key;
+
+            cache 0;
+            cache_fulfill * *;
+            block_action attack reply;
+
+            http_chain {
+                -> tempesta-cat;
+            }
+        """,
+    }
+
+    def test_uri_len(self):
+        """
+        Test 'http_uri_len'.
+
+        Set up `http_uri_len 5;` and make request with uri greater length
+
+        """
+        curl = self.get_client('curl-1')
+
+        self.start_all_servers()
+        self.start_tempesta()
+
+        curl.start()
+        self.wait_while_busy(curl)
+
+        self.assertEqual(
+            self.klog.warn_count(
+                ' Warning: parsed request has been filtered out:',
+            ),
+            ONE,
+        )
+        self.assertEqual(
+            self.klog.warn_count(
+                'Warning: frang: HTTP URI length exceeded for',
+            ),
+            ONE,
+        )
+
+        curl.stop()
+
+    def test_uri_len_without_reaching_the_limit_zero_len(self):
+        """
+        Test 'http_uri_len'.
+
+        Set up `http_uri_len 5;` and make request with uri 0 length
```

Comment:
```suggestion
        Set up `http_uri_len 5;` and make request with uri 1 length
```
There is always something in URI, I believe in this case you have `/` as URI